### PR TITLE
feat: mcap support

### LINKED ISSUE #1: Support mcap datasets
Many robotics platform write miltimodal data in ros2/mcap format where schema is part of the container. Would be amazing to support mcap and read training samples directly from config instead of dumping modalities to disk (images etc)

resources: https://mcap.dev/

### PR DESCRIPTION
**Pull request by egorchakov**
- add mcap readers (protobuf/ros2)
- abstract tabular source processing
- slimmer configs
- add `read-frames` script

closes #1

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
-  "pydantic>=2.8.2",
-  "diskcache",
-  "polars>=1.5.0",
-  "parse",
-  "hydra-core",
-  "simplejpeg",
-  "jaxtyping",
-  "structlog",
+  "tensordict @ git+https://github.com/pytorch/tensordict.git@85b6b81",
+  "torch>=2.4.1",
```
Comment (sandhawalia) [inline pyproject.toml:9]: ❤️

**Comment by sandhawalia** [comment]
successfully build the package and run visualization with carla/yaak/mcap_protobuf. 💪🏽

File: examples/config_templates/table_builder/mcap_ros2.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  _target_: rbyte.io.table.TableMerger
+  separator: /
+  merge:
+    /nissan/vehicle_speed:
+      log_time:
+        method: ref
+
+    /nissan/gps/duro/current_pose:
+      log_time:
+        method: ref
```
Comment (sandhawalia) [inline examples/config_templates/table_builder/mcap_ros2.yaml:32]: This is for which dataset ?

File: src/rbyte/io/table/mcap/ros2_reader.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    # HACK: limit to three levels deep  # noqa: FIX004
+    fields: Mapping[str, Mapping[str, Leaf | Mapping[str, Leaf | Mapping[str, Leaf]]]]
+
+    validate_crcs: bool = False
+
+
+class McapRos2TableReader(TableReaderBase, Hashable):
+    FRAME_INDEX_COLUMN_NAME: Literal["frame_idx"] = "frame_idx"
```
Comment (sandhawalia) [inline src/rbyte/io/table/mcap/ros2_reader.py:41]: This can't be tested in this PR correct ?

**Review by sandhawalia — APPROVED** [review]
Some minor comments. Otherwise LGTM. Build locally and tested it.

File: examples/config_templates/table_builder/mcap_ros2.yaml (hunk 2 above)
Comment (egorchakov) [reply examples/config_templates/table_builder/mcap_ros2.yaml:41]: https://jkk-research.github.io/dataset/jkk_dataset_02/

added link to yaml

File: src/rbyte/io/table/mcap/ros2_reader.py (hunk 3 above)
Comment (egorchakov) [reply src/rbyte/io/table/mcap/ros2_reader.py:27]: i did test this, need an appropriate mcap file tho (https://jkk-research.github.io/dataset/jkk_dataset_02/)

**Pull request merged**